Source/CBL_Attachment.m

```diff
@@ -101,8 +101,13 @@ - (instancetype) initWithName: (NSString*)name
             // Get the revpos:
             id revPosObj = attachInfo[@"revpos"];
             if (revPosObj) {
-                int revPos = [$castIf(NSNumber, revPosObj) intValue];
-                if (revPos <= 0) {
+                if (![revPosObj isKindOfClass: [NSNumber class]]) {
+                    *outStatus = kCBLStatusBadAttachment;
+                    return nil;
+                }
+                int revPos = [revPosObj intValue];
+                // PouchDB has a bug that generates "revpos":0; allow this (#1200)
+                if (revPos < 0) {
```

Comment:
I always has a problem remembering the logic around stub and follows but when we allows revpos = 0, do we also need to change the logic in the line below as well?

https://github.com/couchbase/couchbase-lite-ios/blob/0705423d3d030b208e87f48ee63725c7b553f7ca/Source/CBLDatabase%2BAttachments.m#L257